src/config/mongo.rs

```diff
@@ -0,0 +1,884 @@
+
+    /// Specifies the default read concern for operations performed on the Client. See the
+    /// ReadConcern type documentation for more details.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub read_concern: Option<ReadConcern>,
+
+    /// The name of the replica set that the Client should connect to.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub repl_set_name: Option<String>,
+
+    /// Whether or not the client should retry a read operation if the operation fails.
+    ///
+    /// The default value is true.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub retry_reads: Option<bool>,
+
+    /// Whether or not the client should retry a write operation if the operation fails.
+    ///
+    /// The default value is true.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub retry_writes: Option<bool>,
+
+    /// The default selection criteria for operations performed on the Client. See the
+    /// SelectionCriteria type documentation for more details.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub selection_criteria: Option<ReadPreference>,
+
+    /// The declared API version for this client.
+    /// The declared API version is applied to all commands run through the client, including those
+    /// sent through any handle derived from the client.
+    ///
+    /// Specifying versioned API options in the command document passed to `run_command` AND
+    /// declaring an API version on the client is not supported and is considered undefined
+    /// behaviour. To run any command with a different API version or without declaring one, create
+    /// a separate client that declares the appropriate API version.
+    ///
+    /// For more information, see the [Versioned API](
+    /// https://docs.mongodb.com/v5.0/reference/versioned-api/) manual page.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub server_api: Option<ServerApi>,
+
+    /// The amount of time the Client should attempt to select a server for an operation before
+    /// timing outs
+    ///
+    /// The default value is 30 seconds.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub server_selection_timeout: Option<Duration>,
+
+    /// Default database for this client.
+    ///
+    /// By default, no default database is specified.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub default_database: Option<String>,
+
+    /// The TLS configuration for the Client to use in its connections with the server.
+    ///
+    /// By default, TLS is disabled.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub tls: Option<Tls>,
+
+    /// Specifies the default write concern for operations performed on the Client. See the
+    /// WriteConcern type documentation for more details.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub write_concern: Option<WriteConcern>,
+}
+
+#[allow(missing_docs)]
+impl MongoConfig {
```

Comment:
I wonder if we should use some builder crate in the future

Reply:
I considered using `derive-builder`, but the struct is basically a builder already, so it didn't seem necessary to complicate the access and add another type.